rebase_v1.13.1_action_plan.md

```diff
@@ -0,0 +1,81 @@
+- ✅ Fetch upstream `v1.13.1` and start rebase/merge workflow on latest `origin/main`.
+- ✅ Resolve conflicts, applying consolidation strategy (Section 2).
+- ✅ Re-run `go generate` (`coredns.go`) after reinserting `ocp_dnsnameresolver`.
+- ✅ Verify vendoring strategy (keeping `go mod vendor`; regenerated vendor tree with Go 1.24).
+
+### 5. Build & Test Matrix
+- ✅ Prototype build/tests (Go 1.24) succeed on Linux container via `go build` + `go test ./...`.
+- ⬜️ Once rebase branch compiles, run full downstream test suite (`make test`, e2e suites triggered by CI, QUIC/HTTPS coverage).
+  - ✅ `go test ./...` passes in Linux builder (`podman run --rm --arch=amd64 --user 501:20 -v "/Users/btofel/workspace:/workroot" -w /workroot/coredns -e GOFLAGS=-mod=vendor -e GOTOOLCHAIN=local -e GOCACHE=/tmp/gocache -e GOMODCACHE=/tmp/gomodcache -e HOME=/tmp/home registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.24-openshift-4.21 bash -lc 'mkdir -p /tmp/gocache /tmp/gomodcache /tmp/home && go version && go test ./...'`).
+- ⬜️ Execute targeted plugin validation (route53/Azure/CloudDNS/trace) to confirm upgraded SDK integrations work with OpenShift configs.
+- ⬜️ Run openshift-origin regression/CI rehearsals (prow jobs) with the rebased branch.
+- ✅ Containerized Linux run covers `plugin/dnstap.TestTransport`; keep test execution on Linux hosts for rehearsal runs.
+
+### 6. CI, Build & Packaging Updates
+- 🔄 Determine if downstream Dockerfiles (`Dockerfile.ocp`, `Dockerfile.openshift`) should align with upstream’s distroless changes or remain on ART baseline; document rationale.
+- ⬜️ Refresh `.ci-operator.yaml`, Make targets, and automation metadata once carry consolidation is applied.
+- ⬜️ Update any pipeline references to new Go/tooling requirements (e.g., ensure `GOTOOLCHAIN=local` override is set consistently in CI).
+
+### 7. Release Notes & Communication
+- ⬜️ Maintain a running changelog of notable differences from v1.11.3 → v1.13.1 (QUIC/HTTPS support, dependency bumps, plugin additions, etc.).
+- ⬜️ Capture downstream-specific adjustments (vendoring decisions, carries dropped) for inclusion in automated release notes.
+- ⬜️ Prepare final release notes / upgrade guide draft before merge.
+
+### 8. Post-Rebase Follow-up
+- ⬜️ Remove temporary `replace` once the new plugin version is published and tagged.
+- ⬜️ Archive prototype artifacts (branch can be deleted after merge, document stored alongside reports).
+- ⬜️ Evaluate whether additional automation (e.g., repeating carry audit tooling) should be added for future rebases.
+- ⬜️ Clean up planning artifacts (`carry_consolidation/`, interim reports) once the rebase merges.
+
+## Artifacts & References
+- Rebase report: `rebase_v1.13.1_report.md`
+- Carry audit: `audit_upstream_report.md`
+- Prototype log: `Prototype_Building_v1.13.1_with_OCP_plugin.md`
+- Prototype branch: `prototype/v1.13.1-with-ocp`
+
+_Update this plan as each step completes to keep the rebase effort coordinated and auditable._
```

Comment:
Just confirms my thought that this file looks more like a checklist you may need during the rebase work. I don't see a reason to carry this downstream.

Reply:
agreed